src/com/ms/silverking/cloud/dht/meta/ClassVars.java

```diff
@@ -16,110 +16,110 @@
 
 public class ClassVars implements VersionedDefinition {
-    private final Map<String, String>   classVars;
-    private final long                  version;
-    
-    private static final String     commentEscape = "#";
-    
-    public ClassVars(Map<String, String> classVars, long version) {
-        this.classVars = classVars;
-        this.version = version;
-    }
-    
-    public Map<String,String> getVarMap() {
-        return classVars;
-    }
-    
-    @Override
-    public long getVersion() {
-        return version;
-    }
-    
-    public static ClassVars parse(String def, long version) {
-        try {
-            return parse(new ByteArrayInputStream(def.getBytes()), version);
-        } catch (IOException ioe) {
-            throw new RuntimeException("Unexpected exception", ioe);
-        }
-    }
-    
-    public static ClassVars parse(File fileName, long version) throws IOException {
-        return parse(new FileInputStream(fileName), version);
-    }
-    
-    public static ClassVars parse(InputStream inStream, long version) throws IOException {
-        try {
-            BufferedReader  reader;
-            String          line;
-            Map<String, String>   map;
-            
-            map = new HashMap<>();
-            reader = new BufferedReader(new InputStreamReader(inStream));
-            do {
-                line = reader.readLine();
-                readLine(line, map);
-            } while (line != null);
-            return new ClassVars(map, version);
-        } finally {
-            inStream.close();
-        }
+  private final Map<String, String> classVars;
+  private final long version;
+
+  private static final String commentEscape = "#";
+
+  public ClassVars(Map<String, String> classVars, long version) {
+    this.classVars = classVars;
+    this.version = version;
+  }
+
+  public Map<String, String> getVarMap() {
+    return classVars;
+  }
+
+  @Override
+  public long getVersion() {
+    return version;
+  }
+
+  public static ClassVars parse(String def, long version) {
+    try {
+      return parse(new ByteArrayInputStream(def.getBytes()), version);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Unexpected exception", ioe);
     }
-    
-    private static void readLine(String line, Map<String, String> classVars) {
-        if (line != null) {
-            line = line.trim();
-            if (line.length() > 0 && !line.startsWith(commentEscape)) {
-                String[]    tokens;
-                
-                tokens = line.split("=");
-                if (tokens.length == 2) {
-                    classVars.put(tokens[0], tokens[1]);
-                } else {
-                    throw new RuntimeException("Invalid variable definition: "+ line);
-                }
-            }
-        }
+  }
+
+  public static ClassVars parse(File fileName, long version) throws IOException {
+    return parse(new FileInputStream(fileName), version);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when ClassVars.parse throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).